x-pack/plugins/apm/server/lib/service_nodes/index.ts

```diff
@@ -83,6 +93,12 @@ const getServiceNodes = async ({
       name: bucket.key as string,
       cpu: bucket.cpu.value,
       heapMemory: bucket.heapMemory.value,
+      hostName:
+        bucket.latest.top[0] &&
+        (bucket.latest.top[0].metrics['host.hostname'] as
+          | string
+          | null
+          | undefined),
```

Comment:
Any suggestions in how to improve this part? @sqren @cauemarcondes

Reply:
I think this is okay if these are all possible. Although I'm not sure that both `undefined` and `null` are possible. Can you check what the value is when there are no results?

Reply:
The response is null, when there's no hostname

Reply:
Okay, i think what you've got is correct then.